File: daily/cpp/1422_max_score_after_splitting_a_string.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxScore(string s) {
        int n = s.size();
        int zeros = 0, ones = count(s.begin(), s.end(), '1');
        int res = 0;

        for (int i = 0; i < n - 1; ++i) {
            zeros += (s[i] == '0');
            ones -= (s[i] == '1');
            res = max(res, zeros + ones);
        }
        return res;
    }
};

File: daily/rust/1422_max_score_after_splitting_a_string.rs
impl Solution {
    pub fn max_score(s: String) -> i32 {
        let mut res = 0;
        let mut zeros = 0;
        let mut ones = s.chars().filter(|&c| c == '1').count();

        for c in s.chars().take(s.len() - 1) {
            if c == '0' {
                zeros += 1;
            } else {
                ones -= 1;
            }
            res = res.max(zeros + ones);
        }
        res as i32
    }
}